### ISSUE TITLE
`Segmentation fault(core dumped)` with 0.4.1 on Ubuntu 16.04

### ISSUE THREAD

**Issue by johnhany** [pytorch/pytorch#10430]
## Issue description

Got error `Segmentation fault(core dumped)` with version 0.4.1 on Ubuntu 16.04, even with MNIST example snippets. 

PyTorch was upgraded by `sudo pip3 install --upgrade torch torchvision` from 0.4.0.

I inserted `print(1)` at the [first line](https://github.com/pytorch/examples/blob/master/mnist/main.py#L59) in `main()`, yet only error message was shown and `1` was not printed.

I tried working with Numpy and did not import torch, no error was raised. I swithced back to 0.4.0 and it worked well. So I guess it's related to some upgrades from 0.4.0 to 0.4.1, right;)

## Code example

[MNIST example](https://github.com/pytorch/examples/blob/master/mnist/main.py)

## System Info
PyTorch version: 0.4.1
Is debug build: No
CUDA used to build PyTorch: 9.0.176

OS: Ubuntu 16.04.5 LTS
GCC version: (Ubuntu 5.4.0-6ubuntu1~16.04.10) 5.4.0 20160609
CMake version: version 3.10.3

Python version: 3.5
Is CUDA available: Yes
CUDA runtime version: 9.0.176
GPU models and configuration: GPU 0: GeForce GTX 1080 Ti
Nvidia driver version: 384.130
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.7.0.5
/usr/lib/x86_64-linux-gnu/libcudnn_static_v7.a
/usr/local/lib/python3.5/dist-packages/cntk/libs/libcudnn.so.6

Versions of relevant libraries:
[pip] 18.0

**Comment by ezyang** [comment]
Can you load python with gdb and give us a backtrace?

**Comment by ezyang** [comment]
Leave it to Apport to destroy any useful backtrace information haha.

What happens if you install a fresh Miniconda environment, install torch in it, and try it? I think that should probably fix your problem.

If you want to debug more, can you `ls /usr/local/lib/python3.5/dist-packages/torch/lib/` and tell us what is in that directory?

**Comment by johnhany** [comment]
Well... It happens to conda as well. Here are what I've tried.
1. Install Anaconda with Python 3.6 and PyTorch 0.4.1, run MNIST example and get the following error. (I've disabled Apport)
```
Traceback (most recent call last):
  File "mnist.py", line 3, in <module>
    import torch
  File "/home/john/anaconda3/envs/python35/lib/python3.5/site-packages/torch/__init__.py", line 80, in <module>
    from torch._C import *
ImportError: /home/john/anaconda3/envs/python35/lib/python3.5/site-packages/torch/lib/libshm.so: undefined symbol: _ZTI24THRefcountedMapAllocator
```
2. Install Anaconda with Python 2.7 and PyTorch, get the same `undefined symbol` error.
3. Install Miniconda with Python 3.5 and PyTorch, get the same error:(

And here're all the files in my `/usr/local/lib/python3.5/dist-packages/torch/lib/` directory.
```
include                    libgomp-7bcb08ae.so.1        libshm.so
libcaffe2_gpu.so           libnvrtc-007d19c9.so.9.0     THCUNN.h
libcaffe2.so               libnvrtc-builtins.so         THNN.h
libcudart-f7fdd8d7.so.9.0  libnvToolsExt-3965bdd0.so.1  torch_shm_manager
```
BTW, I'm having some troubles collecting backtrace with gdb when Apport is disabled (by setting `enabled=0` in `/etc/default/apport`). It always shows `no stack` when running command `bt`. If I re-enable Apport, the `py-bt` command does not work properly under the conda environment with gdb, as similar to [this issue](https://github.com/pypa/virtualenv/issues/644) in virtualenv. What should I do to get detailed backtrace information manually?

**Comment by ezyang** [comment]
Can you post the commands you used to install Anaconda/Miniconda and PyTorch? Can you run the Python process with `LD_DEBUG=libs` and post the full log? And finally, can you use `locate libcaffe2.so` to find all copies of this library on your system and report them here?

I don't really know what is going on with Apport, but I don't think it is too relevant

**Comment by johnhany** [comment]
I downloaded `https://repo.anaconda.com/archive/Anaconda3-5.2.0-Linux-x86_64.sh` and run `bash Anaconda3-5.2.0-Linux-x86_64.sh` to install it in `~/anaconda3`. After installing and upgrading several packages, the version of conda is 4.3.30. (To be honest, I'm a little confused by the version number in filename)

I installed PyTorch directly in it by `conda install pytorch torchvision`. Because I live in China, I am using third-party sources for conda and pip provided by tuna.tsinghua.edu.cn. And the downloading speed is really slow if I add `-c pytorch` at the end of the command.

I didn't install a fresh Miniconda environment, but create a minimum environment with `conda create -n python35 python=3.5`. Would it be the same as Miniconda?

I created the Python2.7 environment with `conda create -n python27 python=2.7 anaconda`. Commands that I used to install torch are the same as above.

Also, I have tried:
1. downloaded whl file (Python3.5) directly from `https://pypi.org/project/torch/0.4.1/#files` and installed it. It crashed again.
2. downloaded tar.bz2 files (Python3.6) from `https://anaconda.org/pytorch/pytorch/files` and `https://anaconda.org/pytorch/torchvision/files` and install with `conda install pytorch-0.4.1-py36_cuda9.0.176_cudnn7.1.2_1.tar.bz2` and `conda install torchvision-0.2.1-py36_1.tar.bz2`. It crashed again.

Here are the tailing lines in my `~/.bashrc` file:
```
export GOPATH=$HOME/.go
export PATH=$PATH:/usr/local/cuda/bin:/opt/android-studio/bin:$GOPATH/bin
export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/local/lib:/usr/local/cuda/lib64:/usr/local/cuda/extras/CUPTI/lib64
export PYTHONPATH=$PYTHONPATH:/home/john/Downloads/cleverhans-master

# added by Anaconda3 installer
export PATH="/home/john/anaconda3/bin:$PATH"

alias py27="/usr/bin/python2.7"
alias py35="/usr/bin/python3.5"
```

Here are the full logs when running `LD_DEBUG=libs py35 mnist.py`:
[LD_DEBUG-libs-log.txt](https://gist.github.com/johnhany/9662d051ee8528ee0226f4d913a943d9)
The error occurs at [line 2112](https://gist.github.com/johnhany/9662d051ee8528ee0226f4d913a943d9#file-ld_debug-libs-log-txt-L2112).

And here are the search results for libcaffe2.so:
```
/home/john/Git/caffe2/build/lib/libcaffe2.so
/home/john/anaconda3/envs/python27/lib/python2.7/site-packages/torch/lib/libcaffe2.so
/home/john/anaconda3/lib/python3.6/site-packages/torch/lib/libcaffe2.so
/home/john/anaconda3/pkgs/pytorch-0.4.1-py27_cuda9.0.176_cudnn7.1.2_1/lib/python2.7/site-packages/torch/lib/libcaffe2.so
/home/john/anaconda3/pkgs/pytorch-0.4.1-py35_cuda9.0.176_cudnn7.1.2_1/lib/python3.5/site-packages/torch/lib/libcaffe2.so
/home/john/anaconda3/pkgs/pytorch-0.4.1-py36_cuda9.0.176_cudnn7.1.2_1/lib/python3.6/site-packages/torch/lib/libcaffe2.so
/usr/local/lib/libcaffe2.so
/usr/local/lib/python3.5/dist-packages/torch/lib/libcaffe2.so
```

**Comment by tofudofu** [comment]
Hi, i met the same problem .  it work well on pytorch 0.4.0, but got error on pytorch 0.4.1
here is my commands
conda install pytorch torchvision cuda80 -c pytorch
when i import torch , i got the error:
     from torch._C import *
ImportError: /root/anaconda3/envs/maskrcnn/lib/python3.6/site-packages/torch/lib/libshm.so: undefined symbol: _ZTI24THRefcountedMapAllocator

if you  solve your problems, could you share it with me? 
thanks a lot

**Comment by johnhany** [comment]
Fixed when I uninstall CUDA and Nvidia driver altogether and re-install the latest versions. Now PyTorch 0.4.1 works fine with Nvidia 410.78 driver + CUDA 9.0 (that shipped with PyTorch, while I have installed 9.2).

**Comment by mruberry** [comment]
Closed due to age and apparent fix. If this issue is still occurring please file a new issue with a more recent version of PyTorch.
